src/java/org/apache/cassandra/service/snapshot/TableSnapshot.java

```diff
@@ -325,7 +325,7 @@ static File getLiveFileFromSnapshotFile(Path snapshotFilePath)
         return new File(liveDir.toString(), snapshotFilePath.getFileName().toString());
     }
 
-    public static Predicate<TableSnapshot> shouldClearSnapshot(String tag)
+    public static Predicate<TableSnapshot> shouldClearSnapshot(String tag, long olderThanTimestamp)
```

Comment:
Should we add a specific unit test for this method in addition to the nodetool tests? I think this method is critical enough that deserves a specific unit test to ensure it's doing the right thing (and it will stay that way).

Reply:
added